src/plugins/vis_types/pie/public/editor/components/pie.tsx

```diff
@@ -97,6 +97,9 @@ const PieOptions = (props: PieOptionsProps) => {
   const hasSplitChart = Boolean(aggs?.aggs?.find((agg) => agg.schema === 'split' && agg.enabled));
   const segments = aggs?.aggs?.filter((agg) => agg.schema === 'segment' && agg.enabled) ?? [];
 
+  const legendSize = stateParams.legendSize;
+  const [hadAutoLegendSize, setHadAutoLegendSize] = useState(legendSize === LegendSizes.AUTO);
+
```

Comment:
@flash1293 is there a better way to set `hadAutoLegendSize` once on initial render?

I toyed with the idea of using a `useEffect` hook with an empty dependencies array:
```ts
useEffect(() => {
  setHadAutoLegendSize(legendSize === LegendSizes.AUTO);
}, []);
```

But it seems like that just accomplishes the same thing as using the default argument in `useState`.

But now I'm running into unused-variable errors because the setter isn't used so I didn't know if I was doing something wrong...

Reply:
No one forces you to deconstruct the setter:
```ts
  const [hadAutoLegendSize] = useState(legendSize === LegendSizes.AUTO);
```

also, it doesn't matter in this special case, but you can pass in a constructor function for the initial state as well so it's only executed once:
```ts
  const [hadAutoLegendSize] = useState(() => legendSize === LegendSizes.AUTO);
```